### DISCUSSION TITLE
Fancylists/saneheaders not a "supported" extension

### DISCUSSION THREAD

**Discussion by bakoontz2** [squidfunk/mkdocs-material#7751] [General]
Just curious why fancylists/saneheaders aren't a "supported" extension.  I suspect the reason is because there is a conflict with some other extension?  These seem to be the only two extension that provide any semblance of outlining capability, and they appear to work fine on my install.

I did see one reference to this (#7721) that alluded to some sort of conflict, but it wasn't clear what the conflict might be.

**Comment by facelessuser** [comment]
> Just curious why fancylists/saneheaders aren't a "supported" extension.

I don't think it is stated anywhere that "SaneHeaders" or "FancyLists" are not supported. I literally use it in the documentation for both the plugins (I am the author) and I use MkDocs Material.

I don't know their official stance on these extensions, but they might not be explicitly documented and they may be viewed as optional or even unnecessary extensions by the MkDocs Material team, but there is nothing preventing you from using them.

They may choose to not offer help debugging issues with certain extensions that they deem not critical for the MkDocs Material extensions, but that is understandable as they cannot take time to be knowledgable in all 3rd party extensions.

> I did see one reference to this (https://github.com/squidfunk/mkdocs-material/discussions/7721) that alluded to some sort of conflict, but it wasn't clear what the conflict might be.

I will go on to say Material requires absolutely no formal support for SaneHeaders to work.

I believe there was one issue with FancyLists being that the list type was not honored, hence why you needed to inject styling via the extension options, but I think this was changed via this commit https://github.com/squidfunk/mkdocs-material/commit/43a22b61501c3198519090fd7a26b92f014ca1cc. So it should work, but if it doesn't, inject styles.

As far as I am aware there are no meaningful conflicts preventing usage of these extensions.

**Reply by bakoontz2** [reply]
It's implied at [this link](https://squidfunk.github.io/mkdocs-material/setup/extensions/python-markdown-extensions/), bottom of page:

> The other configuration options of this extension are not officially supported by Material for MkDocs, which is why they may yield unexpected results. Use them at your own risk.

But all that said, your extension is EXACTLY what I was looking for, and I appreciate the time and effort you put into it.  And thank you for responding.

**Reply by squidfunk** [reply]
When users run into problems with anything along the stack, including MkDocs, Python Markdown, Python Markdown Extensions, MkDocs plugins we officially support or any other extension we support, they create an issue __here__. Most users do not bother or do not know to report problems at their respective repositories. Officially supported just means that there is at least some support necessary for an extension to work from Material for MkDocs, and that we commit to maintain the integration and help users to some degree.

We cannot help users with the entirety of the ecosystem, which currently includes [280 plugins](https://github.com/mkdocs/catalog) and probably the same number of Markdown extensions. Hence, the disclaimer. I hope this is understandable from a maintainer's perspective. The problem with MkDocs and the entire ecosystem really is the stack which is spread across many different projects.
